### DISCUSSION TITLE
C# markup like maui?

### DISCUSSION THREAD

**Discussion by Shadowblitz16** [alternetsoft/AlternetUI#128] [Ideas]
Can we have support for C# markup?
xml is really hard to read and without a vscode designer which may take a while it makes it difficult to test alternetui out.

I did try to create my ui with C# but ran into lots of errors and crashes due to me not using the right properties in the right context.

It shouldn't be too hard to support if there is a c# markup library out there that can bind to xml.
If it is too much work I can try to use c# again

**Comment by generalloki** [comment]
I am not sure I understand this issue correctly.  Could you please give more details, provide some examples? How form.uixml, from.g.cs, form.cs should look like?

**Comment by Shadowblitz16** [comment]
@generalloki https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/markup/markup

**Comment by generalloki** [comment]
Idea is great. We will think about it. I personally like c# markup better than xml markup. And to make Alternet.UI c# markup compatible with Maui would be great.  I will post here an update if we decide to work on this.

**Comment by generalloki** [comment]
I am moving this discussions
